crates/re_chunk/src/slice.rs

```diff
@@ -0,0 +1,453 @@
+        // ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+        // │ 5            ┆ -                 ┆ [{x: 3, y: 3}, {x: 4, y: 4}, {x: 5, y: 5}] │
+        // └──────────────┴───────────────────┴────────────────────────────────────────────┘
+        //
+        // The original chunk is unsorted, but the new filtered one actually ends up being sorted.
+        chunk.is_sorted = is_sorted || chunk.is_sorted_uncached();
+
+        #[cfg(debug_assertions)]
+        #[allow(clippy::unwrap_used)] // debug-only
+        chunk.sanity_check().unwrap();
+
+        chunk
+    }
+
+    /// Empties the [`Chunk`] vertically.
+    ///
+    /// The result is a new [`Chunk`] with the same columns but zero rows.
+    #[inline]
+    pub fn emptied(&self) -> Self {
+        let Self {
+            id,
+            entity_path,
+            heap_size_bytes: _,
+            is_sorted: _,
+            row_ids,
+            timelines,
+            components,
+        } = self;
+
+        Self {
+            id: *id,
+            entity_path: entity_path.clone(),
+            heap_size_bytes: Default::default(),
+            is_sorted: true,
+            row_ids: StructArray::new_empty(row_ids.data_type().clone()),
+            timelines: timelines
+                .iter()
+                .map(|(&timeline, time_chunk)| (timeline, time_chunk.emptied()))
+                .collect(),
+            components: components
+                .iter()
+                .map(|(&component_name, list_array)| {
+                    (
+                        component_name,
+                        ListArray::new_empty(list_array.data_type().clone()),
+                    )
+                })
+                .collect(),
+        }
+    }
+}
+
+impl ChunkTimeline {
+    /// Slices the [`ChunkTimeline`] vertically.
+    ///
+    /// The result is a new [`ChunkTimeline`] with the same timelines and (potentially) less rows.
+    ///
+    /// This cannot fail nor panic: `index` and `len` will be capped so that they cannot
+    /// run out of bounds.
+    /// This can result in an empty [`ChunkTimeline`] being returned if the slice is completely OOB.
+    #[inline]
+    pub fn row_sliced(&self, index: usize, len: usize) -> Self {
+        let Self {
+            timeline,
+            times,
+            is_sorted,
+            time_range: _,
+        } = self;
+
+        // NOTE: Bound checking costs are completely dwarfed by everything else, and preventing the
+        // viewer from crashing is more important than anything else in any case.
+
+        if index >= self.num_rows() {
+            return self.emptied();
+        }
+
+        let end_offset = usize::min(index.saturating_add(len), self.num_rows());
+        let len = end_offset.saturating_sub(index);
+
+        if len == 0 {
+            return self.emptied();
+        }
+
+        let is_sorted = *is_sorted || (len < 2);
```

Comment:
This is yet another case where we can guarantee positives but not negatives: gotta take the safe route if `!is_sorted`.